STEP/CodeGeneration/CodeGenerationVisitor.cs

```diff
@@ -10,8 +10,8 @@ public class CodeGenerationVisitor : IVisitor
     private readonly StringBuilder _stringBuilder = new();
     private string Output => _stringBuilder.ToString();
     private readonly StringBuilder _pinSetup = new();
-    private int _scopeLevel = 0;
-    private readonly List<Tuple<int, ArrDclNode>> _arrDclsPerScope = new(); // Keeps track of array declarations per scope
+    private int _scopeLevel;
+    private readonly List<IdNode> _arrDclsInScope = new(); // Keeps track of array declarations currently in scope
```

Comment:
listen beskriver vel alle array dcls? Navnet for det til at lyde som om det er arrays i et specifikt scope

Reply:
Eller listen beskriver vel de aktive array dcls. Altså ja, dem der er in scope...

Reply:
jeps, som du selv skriver i comment 2, arrdlcs der er in scope - altså dem der kan "ses" i programmet på nuværende tidspunkt